sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java

```diff
@@ -732,7 +734,11 @@ synchronized void getTokenAndCallCosmosDbApi(
             TokenExchangeServiceCallback callback) {
         TokenResult cachedTokenResult = TokenManager.getInstance().getCachedToken(partition);
         if (cachedTokenResult != null) {
-            callback.callCosmosDb(cachedTokenResult);
+            if (cachedTokenResult.status().equals(TOKEN_RESULT_SUCCEED)) {
```

Comment:
yeah... is there status check required?

Reply:
Yes - if the token result is not 'succeed', that means that there was some error fetching the token.  You can look at the setup in the test we wrote, let us know if you think there is a problem or this situation is impossible.

Reply:
Was already done in token http callback.